### DISCUSSION TITLE
Negative frequencies in converged SCPH calculations

### DISCUSSION THREAD

**Discussion by zentenmiyata** [ttadano/alamode#139] [Q&A] [answered]
Dear Tadano-sensei,

I am using ALAMODE ver 1.3.0.　I have performed SCPH calculations at T=325 and 326 K, and have checked the convergence as shown below, but at T=326 K, the calculation results contain negative frequencies.　Since the convergence condition states that Ω^2 > 0, I assume that if the calculations converge, there will be no imaginary frequencies, but is this possible?　Incidentally, negative frequency dispersion does not occur at T=325K.

  TMIN = 325; TMAX = 326; DT = 1
  EMIN = 0; EMAX = 1000; DELTA_E = 10
  ISMEAR = -1; EPSILON = 10
  CLASSICAL = 0
  BCONNECT = 0
 Scph:
  KMESH_INTERPOLATE =     1    1    1
  KMESH_SCPH        =     6    6    6
  SELF_OFFDIAG = 0
  IALGO = 0

  RESTART_SCPH = 0
  LOWER_TEMP = 0
  WARMSTART = 1

  TOL_SCPH = 1e-10
  MAXITER = 50000
  MIXALPHA = 0.05

・・・
・・
・

Force constant

  Reading force constants from the XML file ... done !
  Number of non-zero IFCs for 2 order: 125160
  Number of non-zero IFCs for 3 order: 56310
  Number of non-zero IFCs for 4 order: 421728

  Maximum deviation from the translational invariance:
   Order 2 :   8.6392e-16
   Order 3 :   1.2490e-16
   Order 4 :   1.3878e-16

・・・
・・
・

  SCPH ITER   933 :  DIFF =      1.0162e-10
  SCPH ITER   934 :  DIFF =      1.0086e-10
  SCPH ITER   935 :  DIFF =      1.0087e-10
  SCPH ITER   936 :  DIFF =      1.0049e-10
  SCPH ITER   937 :  DIFF =      9.9581e-11
  DIFF < SCPH_TOL : break SCPH loop
 Temp = 3.2500e+02 : convergence achieved in   937 iterations.
Temperature = 3.2600e+02 K
・・・
・・
・

  SCPH ITER  2813 :  DIFF =      1.0668e-10
  SCPH ITER  2814 :  DIFF =      1.0387e-10
  SCPH ITER  2815 :  DIFF =      1.0146e-10
  SCPH ITER  2816 :  DIFF =      9.9141e-11
  DIFF < SCPH_TOL : break SCPH loop
 Temp = 3.2600e+02 : convergence achieved in  2816 iterations.

I would be very grateful if you could tell me how to do this.
Thank you in advance.

Best regards,
Miyata

**Comment by ttadano** [comment]
Even though you set `WARMSTART = 1`, the SCPH took more iterations at 326 K to "converge" than at 325 K, which looks a little strange.

The presence/absence of imaginary modes is checked only for the q points specified by `KMESH_INTERPOLATE`. 
In your setting, it is `1 1 1`, so only the gamma point phonons are examined.

Did you see imaginary branches at the gamma point?
If yes, how close is the phonon frequency of that mode to those of the acoustic phonons (0.0 cm^-1)?

**Reply by zentenmiyata** [reply, marked as answer]
Dear Tadano-sensei,

Thank you very much for your prompt reply.

＞The presence/absence of imaginary modes is checked only for the q points specified by KMESH_INTERPOLATE.

I am sorry for my lack of study. I did not know this fact. As you said, the negative frequency is very close to zero at the Γ point, -7.666702e-06cm^-1 -4.217852e-06 cm^-1 -3.279543e-06 cm^-1.　Negative frequencies of up to -40 cm-1 occur at wavenumbers far from the gamma point.

If the value of KMESH_INTERPOLATE is set too large, the calculation may be forced to terminate in the middle, as if there is not enough memory, so I set KMESH_INTERPOLATE = 1 1 1.

Best regards,
Miyata
